test/functional/tests/incremental_load/test_incremental_load.py

```diff
@@ -358,50 +413,79 @@ def test_print_statistics_inactive(cache_mode):
         second_plug_device = devices["core2"]
+        Udev.disable()  # disabling udev for a while prevents creating clean data on cores
+
     with TestRun.step("Start cache and add cores."):
         cache = casadm.start_cache(cache_dev, cache_mode=cache_mode, force=True)
         first_core = cache.add_core(first_core_dev)
         second_core = cache.add_core(second_core_dev)
+        cache_mode_traits = CacheMode.get_traits(cache.get_cache_mode())
+
+    with TestRun.step("Disable cleaning and sequential cutoff policies."):
+        cache.set_cleaning_policy(CleaningPolicy.nop)
+        cache.set_seq_cutoff_policy(SeqCutOffPolicy.never)
+
     with TestRun.step("Create init config file using current CAS configuration."):
         InitConfig.create_init_config_from_running_configuration()
+
     with TestRun.step("Run IO."):
         run_fio([first_core.system_path, second_core.system_path])
+
     with TestRun.step("Print statistics and check if there is no inactive usage section."):
         active_stats = cache.get_statistics()
         check_if_inactive_section_exists(active_stats, False)
+
     with TestRun.step("Stop cache."):
-        cache.stop()
+        if CacheModeTrait.LazyWrites in cache_mode_traits:
+            cache.stop(no_data_flush=True)
+        else:
+            cache.stop()
+
     with TestRun.step("Remove both core devices from OS."):
+        Udev.enable()  # enable udev back because it's necessary now
         first_plug_device.unplug()
         second_plug_device.unplug()
+
     with TestRun.step("Load cache."):
         cache = casadm.load_cache(cache_dev)
+
     with TestRun.step("Check if inactive devices section appeared and contains appropriate "
                       "information."):
         inactive_stats_before = cache.get_statistics()
         check_if_inactive_section_exists(inactive_stats_before)
         check_number_of_inactive_devices(inactive_stats_before, 2)
+
     with TestRun.step("Attach one of detached core devices and add it to cache."):
         first_plug_device.plug()
+        time.sleep(1)
         first_core_status = first_core.get_status()
         if first_core_status != CoreStatus.active:
             TestRun.fail(f"Core {first_core.system_path} should be in active state but it is not. "
                          f"Actual state: {first_core_status}.")
+
     with TestRun.step("Check cache statistics section of inactive devices."):
         inactive_stats_after = cache.get_statistics()
         check_if_inactive_section_exists(inactive_stats_after)
         check_number_of_inactive_devices(inactive_stats_after, 1)
+        # criteria for checks below
+        insert_write_traits = CacheModeTrait.InsertWrite in cache_mode_traits  # WT, WB, WO
+        lazy_write_traits = CacheModeTrait.LazyWrites in cache_mode_traits  # WB, WO
+        lazy_writes_or_no_insert_write_traits = (not insert_write_traits
+                                                 or lazy_write_traits)  # WA, WB, WO
```

Comment:
Actually using traits has value for being universal, so if we add another cache mode, we won't have to go through all the tests and update conditions accordingly. Using traits and then commenting which cache modes are those, is setting us again in that "search through the code" spot - this time just to update the comments.
Already you've missed PT for 'lazy or no insert' condition ;)

Reply:
You're right, these comments can easily be outdated. I'll get rid of them.